<?xml version="1.0"?>
<launch>
  <!-- Lidar Costmap Generator Node -->
  <node pkg="lattice_planner_morai" type="lidar_costmap_generator" name="lidar_costmap_generator" output="screen">
    
    <!-- Costmap 파라미터 -->
    <param name="map_resolution" value="0.05" />      <!-- 5cm per cell -->
    <param name="map_width" value="50.0" />            <!-- 50m width -->
    <param name="map_height" value="50.0" />           <!-- 50m height -->
    
    <!-- Lidar 필터링 파라미터 -->
    <param name="max_height" value="0.5" />            <!-- Max height to consider (m) -->
    <param name="min_height" value="-0.5" />           <!-- Min height to consider (m) -->
    <param name="lidar_range" value="25.0" />          <!-- Max lidar range (m) -->
    
    <!-- Costmap 값 파라미터 -->
    <param name="obstacle_cost" value="100" />         <!-- Obstacle cell cost -->
    <param name="obstacle_threshold" value="10" />     <!-- Obstacle threshold -->
    
    <!-- MORAI Lidar Topic -->
    <param name="lidar_topic" value="/lidar3D" />
    
    <!-- AABB 박스 마커 토픽 -->
    <param name="marker_topic" value="/pub_cav_aabb_boxes" />
    
  </node>
  
  <!-- RViz visualization (선택사항) -->
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lattice_planner_morai)/config/costmap.rviz" /> -->
</launch>
